sycl/source/detail/platform_impl.cpp

```diff
@@ -49,6 +51,117 @@ platform_impl_host::get_devices(info::device_type dev_type) const {
   return res;
 }
 
+struct DevDescT {
+  const char *devName = nullptr;
+  int devNameSize = 0;
+
+  const char *devDriverVer = nullptr;
+  int devDriverVerSize = 0;
+};
+
+static std::vector<DevDescT> getWhiteListDesc() {
+  // TODO: Replace with const char *str =
+  // SYCLConfig<SYCL_DEVICE_WHITE_LIST>::get();
+  const char *str = getenv("SYCL_DEVICE_WHITE_LIST");
+  if (!str)
```

Comment:
I would like to check this condition as early as possible i.e. before calling `filterWhiteList` to avoid unnecessary overhead.

Reply:
Added additional check before calling filterWhiteList

Reply:
> I would like to check this condition as early as possible i.e. before calling filterWhiteList to avoid unnecessary overhead.

This is already the first operation in `filterWhiteList`, which overhead you are trying to avoid?

Reply:
vector creation + function calls.